packages/tailwindcss/src/design-system.ts

```diff
@@ -82,8 +81,24 @@ export function buildDesignSystem(theme: Theme): DesignSystem {
       return Array.from(parsedVariants.values())
     },
 
-    resolveThemeValue(path: string, defaultValue?: string) {
-      return resolveThemeValue(theme, path, defaultValue)
+    resolveThemeValue(path: `${ThemeKey}` | `${ThemeKey}${string}`) {
```

Comment:
Should we add a comment here that this can be overwritten later by the compatibility implementation? Otherwise, from looking at a stack trace, people might be confused as to why stuff doesn't match up at runtime.